app/uk/gov/hmrc/apiplatformmicroservice/apidefinition/connectors/ApiDefinitionConnector.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apiplatformmicroservice.apidefinition.connectors
+
+import javax.inject.{Inject, Singleton}
+import play.api.Logger
+import uk.gov.hmrc.apiplatformmicroservice.apidefinition.connectors.ApiDefinitionConnector.{ApiDefinitionConnectorConfig, definitionsUrl}
+import uk.gov.hmrc.apiplatformmicroservice.apidefinition.models.APIDefinition
+import uk.gov.hmrc.apiplatformmicroservice.apidefinition.models.JsonFormatters._
+import uk.gov.hmrc.http.{HeaderCarrier, NotFoundException}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+private[apidefinition] class ApiDefinitionConnector @Inject()(http: HttpClient, val config: ApiDefinitionConnectorConfig)
+                                      (implicit val ec: ExecutionContext) {
+
+  def fetchAllApiDefinitions(implicit hc: HeaderCarrier): Future[Seq[APIDefinition]] = {
+    Logger.info(s"${this.getClass.getSimpleName} - fetchAllApiDefinitions")
+    http.GET[Seq[APIDefinition]](definitionsUrl(config.baseUrl), Seq("filterApis" -> "false"))
+  }
+}
+
+private[apidefinition] object ApiDefinitionConnector {
+  case class ApiDefinitionConnectorConfig(baseUrl: String)
```

Comment:
Either name it config or expose it at the top level of the package.

Reply:
What do you mean by top level, the apiplatformmicroservice package? I don't think anything outside the apidefinition package needs to know about the api definition connector.

Reply:
By that I mean within the package - same scope as ApiDefinintionConnector - so basically unwrapping it from the object.  The object is good IF you want to call it ApiDefConnector.Config for example.  What's not nice is ApiDefinitionConnector.ApiDefinitionConfig - though imports mitigate this a bit

Reply:
Ah I see what you mean. Makes sense.